pkg/controllers/user/logging/configsyncer/configsyncer.go

```diff
@@ -0,0 +1,260 @@
+package configsyncer
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/rancher/norman/controller"
+	loggingconfig "github.com/rancher/rancher/pkg/controllers/user/logging/config"
+	"github.com/rancher/rancher/pkg/controllers/user/logging/generator"
+	"github.com/rancher/rancher/pkg/project"
+	"github.com/rancher/types/apis/core/v1"
+	corev1 "github.com/rancher/types/apis/core/v1"
+	mgmtv3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/rancher/types/config/dialer"
+
+	"github.com/pkg/errors"
+	k8scorev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+)
+
+// This controller is responsible for generate fluentd config
+// and updating the config secret
+// so the config reload could detect the file change and reload
+
+func NewConfigSyncer(cluster *config.UserContext, SecretManager *SecretManager, isDeploy *bool) *ConfigSyncer {
+
+	clusterName := cluster.ClusterName
+	return &ConfigSyncer{
+		restConfig:           &cluster.RESTConfig,
+		isDeploy:             isDeploy,
+		clusterName:          clusterName,
+		clusterLoggingLister: cluster.Management.Management.ClusterLoggings(clusterName).Controller().Lister(),
+		projectLoggingLister: cluster.Management.Management.ProjectLoggings(metav1.NamespaceAll).Controller().Lister(),
+		projectLister:        cluster.Management.Management.Projects(clusterName).Controller().Lister(),
+		namespaceLister:      cluster.Core.Namespaces(metav1.NamespaceAll).Controller().Lister(),
+		secretManager:        SecretManager,
+		pods:                 cluster.Core.Pods(loggingconfig.LoggingNamespace),
+		dialer:               cluster.Management.Dialer,
+		clusters:             cluster.Management.Management.Clusters(metav1.NamespaceAll),
+	}
+}
+
+type ConfigSyncer struct {
+	isDeploy             *bool
```

Comment:
Does this bool sharing in `Deployer` and `ConfigSyncer`? It would cause data race problem.